### PR TITLE
Remove unreachable code

### PR DESCRIPTION
**Pull request by GromNaN**
Remove private properties and method never accessed. Found thanks to `./vendor/bin/psalm --find-dead-code`.

This code should be safe to remove, unless it's used for introspection, using reflection or by the C driver.

### CODE REVIEW

File: src/GridFS/WritableStream.php [hunk 1]
```diff
@@ -272,8 +271,6 @@ private function fileCollectionInsert()
 
             throw $e;
         }
-
-        return $this->file['_id'];
```
Comment (GromNaN) [inline src/GridFS/WritableStream.php:276]: Returned value never used.

File: src/Operation/Explain.php [hunk 2]
```diff
@@ -159,13 +159,4 @@ private function createOptions(): array
 
         return $options;
     }
-
-    private function isFindAndModify(Explainable $explainable): bool
```
Comment (jmikola) [inline src/Operation/Explain.php:163]: This was originally introduced in https://github.com/mongodb/mongo-php-library/commit/c6a2df9a3f4c39a2c40181a83fcfbaca66127148 and I missed removing it in https://github.com/mongodb/mongo-php-library/commit/103c1f2ea5c3099a7fccd5b22eabf33fa5eae5a7#diff-fc0bd4b2f7c5d5615660cecd2acc184702ebfbd7153723cc4d0954bc6696bc08L121 when cleaning up code for pre-3.6 servers.

File: src/GridFS/StreamWrapper.php [hunk 3]
```diff
@@ -127,7 +124,6 @@ public function stream_eof(): bool
     public function stream_open(string $path, string $mode, int $options, ?string &$openedPath): bool
     {
         $this->initProtocol($path);
-        $this->mode = $mode;
```
Comment (jmikola) [inline src/GridFS/StreamWrapper.php:130]: Introduced in https://github.com/mongodb/mongo-php-library/commit/e2bf1d45085ab22df90c03d922f1cc99d28d3077#diff-87d70a08d4e98fae61febf5131732ba9564f41ce431a1ea8f4bd759df70c2365R95 and never removed in https://github.com/mongodb/mongo-php-library/commit/bc98d9ccbe30dd7bb4cc4b0ca5104ef3b1d621a2 when `stream_stat()` was refactored.

**Review by jmikola — COMMENTED** [review]
LGTM with the `disableMD5` change reverted (and a new issue created).

Thanks for digging into this.

**Comment by GromNaN** [comment]
Thanks for the review in deep. I reverted `$disableMD5` and created [PHPLIB-1142](https://jira.mongodb.org/browse/PHPLIB-1142)

**Pull request merged**